tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -847,6 +847,96 @@ Valid types are:
     You can turn off this check, or make it only a warning, by changing it in
     the project settings. See :ref:`doc_gdscript_warning_system` for details.
 
+Static variables
+^^^^^^^^^^^^^^^^
+
+A class member variable can be declared static::
+
+    static var a
+
+Local variables cannot be static.
+
+Static variables belong to the class, not instances. This means that static variables
+share values between multiple instances, unlike regular member variables.
+
+From inside a class, you can access static variables from any function, both static and non-static.
+From outside the class, you can access static variables using the class or an instance (not recommended).
+
+::
+
+    # person.gd
+    class_name Person
+
+    static var max_id = 0
+
+    var id
+    var name
+
+    func _init(p_name):
+        max_id += 1
+        id = max_id
+        name = p_name
+
+    func _to_string():
+        return "%s (#%d)" % [name, id]
+
+::
+
+    # test.gd
+    extends Node
+
+    func _ready():
+        var person1 = Person.new("John Doe")
+        var person2 = Person.new("Jane Doe")
+        print(person1) # John Doe (#1)
+        print(person2) # Jane Doe (#2)
+        print(Person.max_id) # 2
+        print(person1.max_id) # 2
+
+Static variables can have type hints and properties::
+
+    static var a: int = 10
+
+    static var b: int:
+        get:
+            return a + 1
+        set(value):
+            a = value - 1
+
+A base class static variable can also be accessed via a child class::
+
+    class A:
+        static var x = 1
+
+    class B extends A:
+        pass
+
+    func _ready():
+        prints(A.x, B.x) # 1 1
+        A.x = 2
+        prints(A.x, B.x) # 2 2
+        B.x = 3
+        prints(A.x, B.x) # 3 3
+
+Static functions can use static member variables. A static variable initializer can call a static function.
```

Comment:
The first sentence is mentioned before. The second sentence means that you can write something like this, right?

```
static var a = some_static_function(value)
```

If so, it'd be good to have a snippet here.

Reply:
I've removed this as it's more related to #7507.